src/auth/hooks.ts

```diff
@@ -0,0 +1,319 @@
+  options?: UseMutationOptions<void, Error>,
+) => {
+  const {authWithOs: enableAuthWithOs, ...mutation} = useAuthWithOs(
+    {
+      ...options,
+      onSuccess: (data, variables, context) => {
+        storage
+          .setItem(SettingsStorageKeys.Auth, JSON.stringify(AUTH_WITH_OS))
+          .then(() => storage.getItem(SettingsStorageKeys.Pin))
+          .then((pin) => (pin != null ? storage.removeItem(SettingsStorageKeys.Pin) : undefined))
+        options?.onSuccess?.(data, variables, context)
+      },
+    },
+    {authenticationPrompt},
+  )
+
+  return {...mutation, enableAuthWithOs}
+}
+
+export const useAuthWithOs = (
+  options?: UseMutationOptions<void, Error>,
+  {authenticationPrompt}: {authenticationPrompt?: AuthenticationPrompt} = {},
+) => {
+  const strings = useStrings()
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const alert = (error: unknown) => {
+    if (error instanceof Keychain.Errors.CancelledByUser) return
+    if (error instanceof Keychain.Errors.TooManyAttempts) return Alert.alert(strings.error, strings.tooManyAttempts)
+    return Alert.alert(strings.error, strings.unknownError)
+  }
+
+  const defaultAuthenticationPrompt: AuthenticationPrompt = {
+    cancel: strings.cancel,
+    title: strings.authorize,
+  }
+
+  const mutation = useMutationWithInvalidations({
+    ...options,
+    invalidateQueries: [['authSetting']],
+    mutationFn: () => Keychain.authenticate(authenticationPrompt ?? defaultAuthenticationPrompt),
+    onError: (error, variables, context) => {
+      alert(error)
+      options?.onError?.(error, variables, context)
+    },
+  })
+
+  return {
+    ...mutation,
+    authWithOs: mutation.mutate,
+  }
+}
+
+export const useAuthOsWithEasyConfirmation = (
+  {id, authenticationPrompt}: {id: YoroiWallet['id']; authenticationPrompt?: AuthenticationPrompt},
+  options?: UseMutationOptions<string, Error>,
+) => {
+  const strings = useStrings()
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const alert = (error: any) => {
+    if (error instanceof Keychain.Errors.CancelledByUser) return
+    if (error instanceof Keychain.Errors.TooManyAttempts) return Alert.alert(strings.error, strings.tooManyAttempts)
+    return Alert.alert(strings.error, strings.unknownError)
+  }
+
+  const primaryAuthenticationPrompt: AuthenticationPrompt = {
+    cancel: strings.cancel,
+    title: strings.authorize,
+  }
+
+  const mutation = useMutation({
+    ...options,
+    mutationFn: () => Keychain.getWalletKey(id, authenticationPrompt ?? primaryAuthenticationPrompt),
+    onError: (error, variables, context) => {
+      alert(error)
+      options?.onError?.(error, variables, context)
+    },
+  })
+
+  return {
+    ...mutation,
+    authWithOs: mutation.mutate,
+  }
+}
+
+export const useDisableEasyConfirmation = (wallet: YoroiWallet, options?: UseMutationOptions) => {
```

Comment:
i think disabling easy confirmation should require authWithOS

Reply:
easy confirmation is per wallet, (it reads the wallet - key), authWithOS, is per App